build/nuke/BuildConfig.cs

```diff
@@ -0,0 +1,166 @@
+                    SourceDirectory.GlobDirectories("**/bin", "**/obj").ForEach(DeleteDirectory);
+                    EnsureCleanDirectory(ArtifactsDirectory);
+                }
+            );
+
+    Target Restore
+        => _ => _
+            .Executes
+            (
+                () =>
+                {
+                    DotNetRestore
+                    (
+                        s => s
+                            .SetProjectFile(Solution)
+                    );
+                }
+            );
+
+    Target Compile => x => Build(x);
+
+    Target Build
+        => _ => _
+            .DependsOn(Restore)
+            .Executes
+            (
+                () =>
+                {
+                    DotNetBuild
+                    (
+                        s => s
+                            .SetProjectFile(Solution)
+                            .SetConfiguration(Configuration)
+                            .SetAssemblyVersion(GitVersion.AssemblySemVer)
+                            .SetFileVersion(GitVersion.AssemblySemFileVer)
+                            .SetInformationalVersion(GitVersion.InformationalVersion)
+                            .EnableNoRestore()
+                    );
+                }
+            );
+
+    Target RegenerateBindings
+        => _ => _
+            .Executes
+            (
+                () =>
+                {
+                    DotNetRun
+                    (
+                        s => s.SetProjectFile(BuildToolsProject)
+                            .SetConfiguration(Configuration)
+                            .SetApplicationArguments(RootDirectory / "generator.json")
+                    );
+                }
+            );
+
+    Target FullBuild
+        => _ => _
+            .DependsOn(RegenerateBindings, Build);
+
+    Target Pack
+        => _ => _
+            .DependsOn(Restore)
+            .Executes
+            (
+                () =>
+                {
+                    DotNetPack
+                    (
+                        s => s
+                            .SetProject(Solution)
+                            .SetConfiguration(Configuration)
+                            .SetAssemblyVersion(GitVersion.AssemblySemVer)
+                            .SetFileVersion(GitVersion.AssemblySemFileVer)
+                            .SetInformationalVersion(GitVersion.InformationalVersion)
+                            .EnableNoRestore()
+                    );
+                }
+            );
+
+    Target FullPack
+        => _ => _
+            .DependsOn(Restore)
+            .DependsOn(RegenerateBindings)
+            .Executes
+            (
+                () =>
+                {
+                    DotNetPack
+                    (
+                        s => s
+                            .SetProject(Solution)
+                            .SetConfiguration(Configuration)
+                            .SetAssemblyVersion(GitVersion.AssemblySemVer)
```

Comment:
Please remove automatic versioning for now.

Reply:
Oh wait hang on I may be mistaken, will this:
- use the tag version on a tag push (i.e. CD builds)
- allow us to override the version number for stuff like CI builds

Reply:
I believe so, yes

Reply:
Are you 100% certain